### PR TITLE
sshd: Make sure service annotation is applied

### PR DESCRIPTION
**Pull request by mbarnes**
This will fix SSHD-controlled load balancer services that are missing the LB connection idle timeout annotation.

(Noticed on some production clusters, since the annotation was added shortly after the initial feature promotion.)

### CODE REVIEW

File: pkg/controller/sshd/sshd_controller.go [hunk 1]
```diff
@@ -324,15 +326,28 @@ func (r *ReconcileSSHD) Reconcile(request reconcile.Request) (reconcile.Result,
...
 		service.Spec.HealthCheckNodePort = foundService.Spec.HealthCheckNodePort
 		if !reflect.DeepEqual(foundService.Spec, service.Spec) {
-			// Specs aren't equal, update and fix.
 			r.SetSSHDStatusPending(instance, "Updating service", "from", foundService.Spec, "to", service.Spec)
 			foundService.Spec = *service.Spec.DeepCopy()
+			serviceNeedsUpdate = true
+		}
+
+		if serviceNeedsUpdate {
 			if err = r.client.Update(context.TODO(), foundService); err != nil {
```
Comment (lisa) [inline pkg/controller/sshd/sshd_controller.go:351]: Is this needed? If the `reflect.DeepEqual` is false, then doesn't it by definition need an update?

Comment (mbarnes) [reply pkg/controller/sshd/sshd_controller.go:351]: It does, but a missing annotation (just above) also sets `serviceNeedsUpdate = true`.

Comment (lisa) [reply pkg/controller/sshd/sshd_controller.go:351]: I see what's happening now. Note to past me: Don't review code when tired!

**Review by lisa — COMMENTED** [review]
/lgtm

**Pull request merged**